src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1722,30 +1722,6 @@ private int tryExecute(final Runnable runnable) {
         TaskNode node = null;
         for (;;) {
             tailNext = tail.getNext();
-            // tail is already consumed
```

Comment:
I would left this check to be executed as the first one, because checking `instance of` force walking the next node until reading the class word, while ref check nope

Reply:
That's a great point, I'll update :-)